Variable type conversion in arguments (#3278)

yamatoRT opened the pull request:
Hi,
This PR implements a part of functionnality of issues #3278 

I implement only type conversion for arguments of keyword.
I reuse same tests of python function annotations, except for the vargs and the kwargs. Moreover, i remove tests for decorators.
Don't hesitate to challenge the name of class or method, the english is not my  best language.

Best Regards,

miktuy commented on atest/testdata/keywords/type_conversion/RobotKeywordAnnotationsWithAliases.robot:
```diff
@@ -0,0 +1,87 @@
+    Validate Type     ${argument}    ${expected}
+
+Boolean
+    [Arguments]    ${argument: 'Boolean' }    ${expected}=${Empty}
```
extra space after ```'Boolean'``` here and in line 34

yamatoRT replied:
I removed extra space.

miktuy commented on src/robot/running/arguments/argumentparser.py:
```diff
@@ -16,9 +16,11 @@
 from robot.errors import DataError
 from robot.utils import JYTHON, PY_VERSION, PY2
 from robot.variables import is_dict_var, is_list_var, is_scalar_var
+from pydoc import locate
```
PEP8 import order
```
Imports should be grouped in the following order:

Standard library imports.
Related third party imports.
Local application/library specific imports.
You should put a blank line between each group of imports.
```

yamatoRT replied:
Thank for comments. I change import order

pekkaklarck commented:
First of all, thanks for the PR! This is pretty big task to implement and I've been thinking it's too big for the already overdue RF 3.2, but including it is still possible if we can get this PR into a good enough shape. I won't have time for a full review until I have got RF 3.2 alpha 1 ready, but below are some general comments about this PR after a very quick look:

1. This PR only adds support for argument conversion. Conversion in the Variables section and also when assigning variables based on keyword return values are needed as well. Probably a good idea to get this first feature ready first.

2. I don't think we need to support syntax like `${arg: 'int'}`. Just supporting `${arg: int}` (space being optional) ought to be enough. I also don't see reasons why to support Python's `typing` style usages like `${arg:Mapping[str, int]}` either. Just supporting same string names and aliases that work with the dynamic library API (#2068) ought to be enough.

3. Implementation looks pretty complicated. Most probably that's due to the support for formats like `${arg:Mapping[str, int]}`. I think this could be done pretty easily in [argumentparser.py](https://github.com/robotframework/robotframework/blob/master/src/robot/running/arguments/argumentparser.py): `PythonArgumentParser` already does it with Python based keywords and `UserKeywordArgumentParser` could probably do it with user keywords.

yamatoRT commented:
Hi,
If we don't need the typing style usages, the code will be simplied.
I will rework my code with your comments.

yamatoRT commented:
I removed typing style usages, abc types, enum and bytesstring type. The other types is already convertible by **TypeConverter**.
Too I put my implementation in  **argumentparser.py** and I removed **typingresolver.py**.

yamatoRT commented:
I am closing this PR because i will propose new implementation for the issue #3278 who work with variable
